src/Compilers/CSharp/Portable/Symbols/AssemblySymbol.cs

```diff
@@ -858,11 +866,31 @@ private NamedTypeSymbol ApplyGenericArguments(NamedTypeSymbol symbol, Type[] typ
                             continue;
                         }
                     }
+                    else if (isWellKnownType && warnings is null)
+                    {
+                        // For well-known types after C# 7, we prefer corlib (unless ignoreCorLibraryDuplicatedTypes is set)
+                        if (IsInCorLib(result))
+                        {
+                            // ignore candidate
+                            continue;
+                        }
+                        if (IsInCorLib(candidate))
+                        {
+                            // drop previous result
+                            result = candidate;
+                            continue;
+                        }
+                    }
 
-                    if (warnings == null)
+                    if (warnings is null)
                     {
                         conflicts = (result.ContainingAssembly, candidate.ContainingAssembly);
                         result = null;
+                        if (isWellKnownType)
+                        {
+                            onlyConsiderCorLib = true;
```

Comment:
>onlyConsiderCorLib = true; [](start = 28, length = 26)

It feels like this would conflict with ```ignoreCorLibraryDuplicatedTypes``` flag when it is true.